notebooks/source/ordinalregression.ipynb

```diff
@@ -0,0 +1,239 @@
+      "text/plain": [
+       "<Figure size 432x288 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "sns.violinplot(x='Y', y='X', data=df)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We will model the outcomes Y as coming from an OrderedLogistic distribution, conditional on X. The `OrderedLogistic` distribution in numpyro requires ordered cutpoints. We can use the `ImproperUnifrom` distribution to introduce a parameter with an arbitrary support that is otherwise completely uninformative, and then add an `ordered_vector` constraint."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "sample: 100%|██████████| 1000/1000 [00:02<00:00, 341.47it/s, 7 steps of size 4.34e-01. acc. prob=0.95]\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n",
+      "                mean       std    median      5.0%     95.0%     n_eff     r_hat\n",
+      "   b_X_eta      1.43      0.35      1.42      0.82      1.96    352.72      1.00\n",
+      "    c_y[0]     -0.11      0.41     -0.11     -0.78      0.55    505.85      1.00\n",
+      "    c_y[1]      2.18      0.52      2.15      1.35      2.95    415.22      1.00\n",
+      "\n",
+      "Number of divergences: 0\n"
+     ]
+    }
+   ],
+   "source": [
+    "def model1(X, Y, nclasses=3):\n",
+    "    b_X_eta = sample('b_X_eta', Normal(0, 5))\n",
+    "    c_y     = sample('c_y',     ImproperUniform(support=constraints.ordered_vector,\n",
+    "                                                batch_shape=(),\n",
+    "                                                event_shape=(nclasses-1,)))\n",
+    "    with numpyro.plate('obs', X.shape[0]):\n",
+    "        eta = X * b_X_eta\n",
+    "        sample('Y', OrderedLogistic(eta, c_y), obs=Y)\n",
+    "\n",
+    "mcmc_key = random.PRNGKey(1234)\n",
+    "kernel = NUTS(model1)\n",
+    "mcmc   = MCMC(kernel, num_warmup=250, num_samples=750)\n",
+    "mcmc.run(mcmc_key, X,Y, nclasses)\n",
+    "mcmc.print_summary()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If we have additional information on the parameters that we want to encode in a prior, we can add a `sample` statement that uses a 'regular' prior with an `obs` argument, conditioning the parameter drawn from the `ImproperUniform` on the chosen prior"
```

Comment:
It would be great if you explain this approach a little bit more. I don't have much idea but something like mentioning that this is equivalent to using an "unnormalized" prior (created by restricting the support of a Normal distribution to the "ordered" domain). :)

Reply:
I've tried to elaborate on this a bit. Is is better phrased right now? Do you have additional suggestions?

Reply:
Thanks, @vanAmsterdam! It look great to me.